### PR TITLE
[HZ-67] hz-5.0 updates

### PR DESCRIPTION
**Pull request by fbarotov**

### CODE REVIEW

File: distro/Makefile
Comment (frant-hartm) [inline distro/Makefile:34]: This url is correct for snapshot, this script will run also for release version, we need to check the HZ_VERSION and if it doesn't contain `-SNAPSHOT` it should use `https://repo1.maven.org/maven2/`

File: distro/Makefile
Comment (frant-hartm) [inline distro/Makefile:45]: We are removing hazelcast-all in 5.0, so the scripts should be updated to use hazelcast-${HZ_VERSION}.jar.

File: packages/deb/hazelcast/DEBIAN/control
Comment (frant-hartm) [inline packages/deb/hazelcast/DEBIAN/control:3]: I think the version should be something like `5.2021.8` (8 for August release). Don't ask me why :-)

**Review by frant-hartm — CHANGES_REQUESTED** [review]

**Review by alparslanavci — CHANGES_REQUESTED** [review]
If this PR is for releasing Hazelcast's `SNAPSHOT` versions in package managers, there should be some more changes. First, the package repos should be different for `SNAPSHOT` releases, the user should specify this repo if they would like to install `SNAPSHOT` release. Otherwise, `SNAPSHOT`s can be installed to prod environments mistakenly.

The GH action files also need to be updated accordingly, just changing the versions in the master branch won't help to install `SNAPSHOT` versions.

**Comment by fbarotov** [comment]
> If this PR is for releasing Hazelcast's `SNAPSHOT` versions in package managers, there should be some more changes. First, the package repos should be different for `SNAPSHOT` releases, the user should specify this repo if they would like to install `SNAPSHOT` release. Otherwise, `SNAPSHOT`s can be installed to prod environments mistakenly.
> 
> The GH action files also need to be updated accordingly, just changing the versions in the master branch won't help to install `SNAPSHOT` versions.

thanks alparslanavci :) just learned, no we are not releasing SNAPSHOT versions yet, removing SNAPTHOT stuff ...

File: packages/deb/hazelcast/DEBIAN/control
Comment (fbarotov) [reply packages/deb/hazelcast/DEBIAN/control:3]: Hmm, thanks. I am not sure, but this change should be made after the full release probably. The reason I went with snapshot is to pass integration tests and thus, make adjustments according to Hazelcast 5.0 changes.

**Comment by alparslanavci** [comment]
@fbarotov, the `integration-tests` GH actions are failing, could you please check them?

**Comment by fbarotov** [comment]
@alparslanavci ha yeah, it is kinda expected because here we have hazelcast version set as `5.2021.8`, so we want this PR to get approved but will not merge it until Hazelcast 5.0 is released

**Comment by mmedenjak** [comment]
Agree with @fbarotov , I'd like to be sure this PR is looking good and there are no concerns from the reviewer perspective, but ultimately we'll need to re-test once the new artifacts are released.

File: packages/deb/hazelcast/DEBIAN/conffiles
Comment (alparslanavci) [inline packages/deb/hazelcast/DEBIAN/conffiles:1]: @fbarotov, what is the reason for removing these?

File: packages/deb/hazelcast/DEBIAN/control
Comment (alparslanavci) [inline packages/deb/hazelcast/DEBIAN/control:3]: I don't think you need to change these versions. They are changed by the GH action jobs when releasing.

File: packages/deb/hazelcast/DEBIAN/postinst
Comment (alparslanavci) [inline packages/deb/hazelcast/DEBIAN/postinst:2]: Same for these. They are changed by the GH action jobs when releasing.

File: packages/rpm/hazelcast.spec
Comment (alparslanavci) [inline packages/rpm/hazelcast.spec:4]: Same for these versions. Please see my comment above.

File: packages/rpm/hazelcast.spec
Comment (alparslanavci) [inline packages/rpm/hazelcast.spec:41]: This is a bit interesting. By adding `/bin` folder to the path, there will be four commands available to the users: `hz`, `hazelcast`, `hazelcast-start`, and `hazelcast-stop`. All of them have different (some overlapping) purposes. Maybe @Holmistr can comment on that.

File: pom.xml
Comment (alparslanavci) [inline pom.xml:13]: Having a non-released version in the master branch is not good. Maybe we can merge this into a maintenance branch (such as `5.0-snapshot`) and merge it with `master` when Hazelcast 5.0 is released.

**Review by alparslanavci — CHANGES_REQUESTED** [review]
Thanks for the PR, @fbarotov. I also added it as a comment, but better to mention it again here. IMO, having a non-released version in the master branch is not good. Maybe we can merge this into a maintenance branch (such as 5.0-snapshot) and merge it with `master` when Hazelcast 5.0 is released. Otherwise, the tests will continue to fail until 5.0 is released.

File: pom.xml
Comment (fbarotov) [reply pom.xml:13]: thanks :) created `5.0-snapshot` branch and changed base branch of this PR

File: packages/rpm/hazelcast.spec
Comment (Holmistr) [reply packages/rpm/hazelcast.spec:41]: I know that this might be a bit confusing but we can't do any better. They all serve a different purpose, so we want to keep them there.
But it's definitely a room for improvement in the next releases.

File: packages/deb/hazelcast/DEBIAN/conffiles
Comment (fbarotov) [reply packages/deb/hazelcast/DEBIAN/conffiles:1]: In Hazelcast 5.0 distribution these xml/yaml files will not be under `bin/` directory. There is a PR on consolidating all configuration related files under `examples/` directory: https://github.com/hazelcast/hazelcast/issues/18990

After that PR is merged we can apply updates here as well.

Comment (alparslanavci) [reply packages/deb/hazelcast/DEBIAN/conffiles:1]: @fbarotov these files are considered as user config files and are marked in `conffiles` to prevent overwriting when a new version is installed. A similar marking exists in the `hazelcast.spec` file as `%config(noreplace)` keyword for RPM packages.

**Review by alparslanavci — COMMENTED** [review]
Added a single comment, otherwise LGTM 👍

File: packages/deb/hazelcast/DEBIAN/conffiles
Comment (fbarotov) [reply packages/deb/hazelcast/DEBIAN/conffiles]: Ohh, I see, thank you for the explanation

**Review by frant-hartm — APPROVED** [review]

**Review by alparslanavci — APPROVED** [review]
@fbarotov, approving this PR, but most likely `release.yml` should be also updated to fix the issues with the user config files. Please have a look it before merging this branch with the master.

**Comment by fbarotov** [comment]
@alparslanavci thank you for the review and approval. I updated `release.yml` to accommodate changes on configuration example path in Hazelcast 5.0

**Comment by Holmistr** [comment]
@alparslanavci @fbarotov Guys, I'm confused. Shouldn't this be merged before the Beta was out? So it won't work?

**Comment by mmedenjak** [comment]
As explained on slack and just for reference, we were not able to merge this before the actual HZ release because the code needs deployed artifacts in public repos. For instance, this comment from Alparslan:

> I also added it as a comment, but better to mention it again here. IMO, having a non-released version in the master branch is not good. Maybe we can merge this into a maintenance branch (such as 5.0-snapshot) and merge it with master when Hazelcast 5.0 is released. Otherwise, the tests will continue to fail until 5.0 is released.

**Comment by mmedenjak** [comment]
Thank you everyone!

**Pull request merged**
